R/zchunk_LA118.hydro.R

```diff
@@ -29,41 +29,177 @@ module_energy_LA118.hydro_DISABLED <- function(command, ...) {
     # Load required inputs
     iso_GCAM_regID <- get_data(all_data, "common/iso_GCAM_regID")
     Hydropower_potential <- get_data(all_data, "energy/Hydropower_potential")
-    L100.IEA_en_bal_ctry_hist <- get_data(all_data, "L100.IEA_en_bal_ctry_hist")
+
+    get_data(all_data, "temp-data-inject/L100.IEA_en_bal_ctry_hist") %>%
+      gather(year, value, -iso, -FLOW, -PRODUCT) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) ->
+      L100.IEA_en_bal_ctry_hist
+
     A18.hydro_output <- get_data(all_data, "energy/A18.hydro_output")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+
+    # Calculation of economic hydropower potential by country, in EJ/yr
+    # Calculate a capacity factor for translating MW to GWh, using weighted average capacity factor of all existing dams
+    Hydropower_potential %>%
+      gather(variable, value, -Country, -iso) %>%
+      group_by(variable) %>%
+      summarise(value = sum(value, na.rm = TRUE)) %>%
+      # Calculation of capacity factor
+      summarise(value = value[variable == "Installed_GWh"] / (value[variable == "Installed_MW"] * CONV_YEAR_HOURS * CONV_MIL_BIL)) %>%
```

Comment:
I agree with Ben.  This is a weird use of `summarise`, especially since the data frame is grouped by variable.  What is the intended result of line 49?  That is, after line 49, what do you expect the data to look like?